packages/kbn-scout-reporting/src/reporting/playwright.ts

```diff
@@ -0,0 +1,265 @@
+        type: 'playwright',
+      },
+      test_run: {
+        id: this.runId,
+      },
+      suite: {
+        title: test.parent.titlePath().join(' '),
+        type: test.parent.type,
+      },
+      test: {
+        id: getTestIDForTitle(test.titlePath().join(' ')),
+        title: test.title,
+        tags: test.tags,
+        annotations: test.annotations,
+        expected_status: test.expectedStatus,
+        step: {
+          title: step.titlePath().join(' '),
+          category: step.category,
+        },
+      },
+      event: {
+        action: ScoutReportEventAction.TEST_STEP_BEGIN,
+      },
+    });
+  }
+
+  onStepEnd(test: TestCase, result: TestResult, step: TestStep) {
+    this.report.logEvent({
+      ...environmentMetadata,
+      reporter: {
+        name: this.name,
+        type: 'playwright',
+      },
+      test_run: {
+        id: this.runId,
+      },
+      suite: {
+        title: test.parent.titlePath().join(' '),
+        type: test.parent.type,
+      },
+      test: {
+        id: getTestIDForTitle(test.titlePath().join(' ')),
+        title: test.title,
+        tags: test.tags,
+        annotations: test.annotations,
+        expected_status: test.expectedStatus,
+        step: {
+          title: step.titlePath().join(' '),
+          category: step.category,
+          duration: step.duration,
+        },
+      },
+      event: {
+        action: ScoutReportEventAction.TEST_STEP_END,
+        error: {
+          message: step.error?.message,
+          stack_trace: step.error?.stack,
+        },
+      },
+    });
+  }
+
+  onTestEnd(test: TestCase, result: TestResult) {
+    this.report.logEvent({
+      ...environmentMetadata,
+      reporter: {
+        name: this.name,
+        type: 'playwright',
+      },
+      test_run: {
+        id: this.runId,
+      },
+      suite: {
+        title: test.parent.titlePath().join(' '),
+        type: test.parent.type,
+      },
+      test: {
+        id: getTestIDForTitle(test.titlePath().join(' ')),
+        title: test.title,
+        tags: test.tags,
+        annotations: test.annotations,
+        expected_status: test.expectedStatus,
+        status: result.status,
+        duration: result.duration,
+      },
+      event: {
+        action: ScoutReportEventAction.TEST_END,
+        error: {
+          message: result.error?.message,
+          stack_trace: result.error?.stack,
+        },
+      },
+    });
+  }
+
+  onEnd(result: FullResult) {
+    this.report.logEvent({
+      ...environmentMetadata,
+      reporter: {
+        name: this.name,
+        type: 'playwright',
+      },
+      test_run: {
+        id: this.runId,
+        status: result.status,
+        duration: result.duration,
+      },
+      event: {
+        action: ScoutReportEventAction.RUN_END,
+      },
+    });
+
+    // Save & conclude the report
+    try {
+      this.report.save(this.reportRootPath);
+    } finally {
+      this.report.conclude();
+    }
```

Comment:
If I read this correctly, we would throw away the report if `save` doesn't work for some reason. Do you think it would be worth to keep it in that case for a later retry or manual access?

Reply:
Any attempt to resolve an issue that occurs during `save` would have to happen in the same session. The reporter is operating in a temporary directory for safety during cleanup. I can't imagine a scenario where keeping the temporary work directory would help.